content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb

```diff
@@ -195,7 +195,38 @@ def collect_totals(array)
 
 def collect_dialog_totals(prov_option, options_hash)
   dialog_values(prov_option, options_hash, dialog_array = [])
-  collect_totals(dialog_array)
+  total = collect_totals(dialog_array)
+  if prov_option == :number_of_cpus
+    options_hash.each do |_sequence_id, options|
+      if options.keys.include?(:number_of_sockets) || options.keys.include?(:cores_per_socket)
+        $evm.log(:info, "Recalculating number of cpus based on dialog overrides")
+        total = recalculate_number_of_cpus(service_resource, options[:number_of_sockets].to_i, options[:cores_per_socket].to_i)
+      end
+    end
+  end
+  total
+end
+
+def service_resource
+  resource = nil
+  @service_template.service_resources.each do |child_service_resource|
+    next if @service_template.service_type == 'composite'
+    next if @service_template.prov_type.starts_with?("generic")
+    resource = child_service_resource.resource
+  end
+  resource
+end
+
+def recalculate_number_of_cpus(resource, override_number_of_sockets, override_cores_per_socket)
+  $evm.log(:info, "Recalculating the number_of_cpus resource: #{resource} override_number_of_sockets: #{override_number_of_sockets} override_cores_per_socket: #{override_cores_per_socket}")
+  if override_number_of_sockets.positive? && override_cores_per_socket.positive?
```

Comment:
@billfitzgerald0120 Is the positive check because a user can enter a negative value or is it to protect from a user not entering any value and it stays nil?

Reply:
It is to protect from an user not entering a value